core/ai.go

```diff
@@ -23,14 +24,40 @@ type AI interface {
 	HasCapacity(pipeline, modelID string) bool
 }
 
+// Custom type to handle both string and int but store as string.
+type StringInt string
+
+// UnmarshalJSON method to handle both string and int.
+func (s *StringInt) UnmarshalJSON(data []byte) error {
+	// Try to unmarshal as int.
+	var intValue int64
+	if err := json.Unmarshal(data, &intValue); err == nil {
+		*s = StringInt(strconv.FormatInt(intValue, 10))
+		return nil
+	}
+
+	var strValue string
+	if err := json.Unmarshal(data, &strValue); err == nil {
+		*s = StringInt(strValue)
+		return nil
+	}
+
+	return fmt.Errorf("invalid value for StringInt: %s", data)
+}
+
+// String converts the StringInt type to a string.
+func (s StringInt) String() string {
+	return string(s)
+}
+
 type AIModelConfig struct {
 	Pipeline          string                   `json:"pipeline"`
 	ModelID           string                   `json:"model_id"`
 	URL               string                   `json:"url,omitempty"`
 	Token             string                   `json:"token,omitempty"`
 	Warm              bool                     `json:"warm,omitempty"`
-	PricePerUnit      int64                    `json:"price_per_unit,omitempty"`
-	PixelsPerUnit     int64                    `json:"pixels_per_unit,omitempty"`
+	PricePerUnit      StringInt                `json:"price_per_unit,omitempty"`
+	PixelsPerUnit     StringInt                `json:"pixels_per_unit,omitempty"`
```

Comment:
Instead of making these strings, I think you should implement a similar logic to this: https://github.com/livepeer/go-livepeer/pull/2981/files#diff-8e2167038b87d117b916131f62552da8dee7f52d8112e490330349597b393617R1489-R1490

We can't parse these values to an `int` as sometimes they can be a `float` (like `0.20` dollars), but we also can't parse it to a `float` cause we lose precision. So the idea there, which I believe should be replicated here, is to keep it as the raw characters that were in the JSON (`json.RawMessage`) and then parsing it as `big.Rat` 🐀 

One thing you could do to make this less repetitive would be creating a custom type to do exactly that. Not to support a `string` field (the currency should go on a separate field IMO, as per the other comment), but to parse it raw into a `big.Rat` (you don't even need a `json.RawMessage` confusion in that case, it should be pretty straightforward). With that custom type we could use it both here and in the per-gateway pricing config.

WDYT?

Reply:
I took the liberty of implementing that and pushing to the branch here, since it was something I already wanted to do and you have a lot on your plate with this merge rn.

Here's the commit: https://github.com/livepeer/go-livepeer/pull/3103/commits/8036856cbfd6724f79b393118e49fd2451282797

Lmk what you think and feel free to revert it if you'd rather go a different direction. Notice that I haven't fixed the currency field on that commit, need to add that as well.

Reply:
This is a great change. Thanks for implementing it. I will implement the `Currency` field 👍🏻.